### PR TITLE
loss must be avg when BS>1 when calling evaluate_batch()

### PR DESCRIPTION
**Pull request by sangeet2020**
Hi @ycemsubakan ,

I initiate this PR to fix a bug in the `WHAMandWHAMR` recipe. I realized that the current recipe works fine on batch_size as 1. However, when increased, losses are not averaged when `evaluate_batch` is called. 
Basically, it makes sense to write
```python
loss = self.compute_objectives(predictions, targets)
loss = torch.mean(loss)   # Avg losses in a batch
```

Thank You

### CODE REVIEW

**Comment by mravanelli** [comment]
@ycemsubakan could you please take a look?

**Comment by ycemsubakan** [comment]
@mravanelli @sangeet2020 I will try to do it this week.

**Comment by MartinKocour** [comment]
@sangeet2020 take a look into mine LibriCSS recipe. IMHO, it should be done inside compute_objective
LibriCSS recipe is here:
https://github.com/MartinKocour/speechbrain/blob/49ab1668c8e8952cf3f33efa3d8d1f15cd3f42a4/recipes/LibriCSS/separation/train.py#L139

**Comment by ycemsubakan** [comment]
@MartinKocour , thanks for pointing this out. I 'll take a look tonight.

**Comment by sangeet2020** [comment]
> @sangeet2020 take a look into mine LibriCSS recipe. IMHO, it should be done inside compute_objective LibriCSS recipe is here: https://github.com/MartinKocour/speechbrain/blob/49ab1668c8e8952cf3f33efa3d8d1f15cd3f42a4/recipes/LibriCSS/separation/train.py#L139

yes, this is a suitable choice. I will modify it and re-commit. thanks @MartinKocour

**Comment by ycemsubakan** [comment]
Hmm... we need make sure it works with both frequency domain and time.. I am checking now..

**Comment by ycemsubakan** [comment]
I pushed a small cosmetic update. @sangeet2020 , are you planning to have more updates? If yes, we can keep this PR open. If not, we can merge.

**Review by ycemsubakan — APPROVED** [review]

**Comment by ycemsubakan** [comment]
> Hi Cem, I dont plan any more updates on this PR, I think we are safe to close and merge it.

Alright, thanks for the PR @sangeet2020!
